Update to v10.0.0

rxm7706 opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->
closes https://github.com/conda-forge/ibis-framework-feedstock/pull/108
closes https://github.com/conda-forge/ibis-framework-feedstock/pull/110
<!--
Please add any other relevant info below:
-->

- Removed: ibis-pandas backend
- Added: ibis-athena and ibis-databricks backend
- Deleted: recipe/conda_build_config.yaml used {% set python_min = "3.10" %} in meta.yaml instead
- Fixed dependencies:
  - based on https://github.com/ibis-project/ibis/blob/10.0.0/pyproject.toml
  - deleted multipledispatch fro ibis-framework-core
  - updated rich >=12.4.4,<14 in ibis-framework-core
  - updated sqlglot >=23.4,<26.4 in ibis-framework-core
- Added upper limits to outputs deps as needed
  - psycopg2 >=2.8.4,<3 for ibis-risingwave
  - pydata-google-auth >=1.4.0,<2 for ibis-bigquery
  - pins >=0.8.3,<1 for ibis-duckdb
  - psycopg >=3.2.0,<4 for ibis-postgres
- Add pip check to all outputs
- Fixed linter issues by adding noarch python {{ python_min }} to host for all outputs
- Fixed linter issues by adding noarch python >={{ python_min }} to run for all outputs
- Fixed sequence / minor formatting updates to all outputs

Pending
- https://github.com/conda-forge/admin-requests/pull/1383

rxm7706 committed:
Delete recipe/conda_build_config.yaml

rxm7706 committed:
Update to v10.0.0

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of `<output 1` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 2` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 2` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 3` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 3` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 4` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 4` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 5` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 5` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 6` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 6` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 7` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 7` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 8` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 8` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 9` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 9` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 10` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 10` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 11` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 11` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 12` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 12` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 13` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 13` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 14` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 14` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 15` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 15` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 16` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 16` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 17` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 17` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 18` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 18` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 19` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 19` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13231127953](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13231127953). Examine the logs at this URL for more detail.</sub>

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.09.14.38.22

rxm7706 committed:
Remove ibis.backends.pandas

rxm7706 committed:
Fix ibis-postgres dep

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13231159381](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13231159381). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
pip checks

rxm7706 committed:
Please the Linter

rxm7706 committed:
Merge pull request #1 from rxm7706/v10.0.0-Base-PipCheck

V10.0.0 base pip check

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13242994879](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13242994879). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Revert PiP Checks

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of `<output 1` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 2` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 2` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 3` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 3` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 4` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 4` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 5` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 5` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 6` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 6` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 7` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 7` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 8` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 8` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 9` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 9` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 10` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 10` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 11` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 11` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 12` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 12` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 13` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 13` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 14` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 14` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 15` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 15` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 16` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 16` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 17` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 17` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 18` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 18` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 19` output, you should usually use `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 19` output, you should usually use `python >={{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13243456420](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13243456420). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13243462353](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13243462353). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Fixes

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247523872](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247523872). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Remove pip check

rxm7706 commented:
@conda-forge-admin, please rerender

rxm7706 committed:
fix

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247738842](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247738842). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
fixes

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247824725](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13247824725). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Add PiP Checks to all packages

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13251671454](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13251671454). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Add requires pip for pip check

cpcloud commented:
@rxm7706 Is this ready for merge?

rxm7706 committed:
Add ibis-athena, Ibis-databricks

rxm7706 commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.11.19.34.16

rxm7706 committed:
Fix Databricks connector name

rxm7706 commented:
to do - add Output Validation and Feedstock Token for new outputs - ibis-athena ibis-databricks
https://conda-forge.org/docs/maintainer/infrastructure/#output-validation-and-feedstock-tokens
''validation results:
{
  "noarch/ibis-athena-10.0.0-pyhd8ed1ab_0.conda": false,
  "noarch/ibis-flink-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-mysql-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-polars-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-impala-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-trino-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-databricks-10.0.0-pyhd8ed1ab_0.conda": false,
  "noarch/ibis-duckdb-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-framework-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-sqlite-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-pyspark-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-clickhouse-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-exasol-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-snowflake-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-mssql-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-postgres-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-oracle-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-risingwave-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-druid-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-framework-core-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-bigquery-10.0.0-pyhd8ed1ab_0.conda": true,
  "noarch/ibis-datafusion-10.0.0-pyhd8ed1ab_0.conda": true
}
NOTE: Any outputs marked as False are not allowed for this feedstock. See https://conda-forge.org/docs/maintainer/infrastructure/#output-validation-and-feedstock-tokens for information on how to address this error.
''

rxm7706 commented:
> @rxm7706 Is this ready for merge?

hi @cpcloud - 
ready for a review, and comments, but not yet for merge, I have updated the description of the PR to help. 
Updated the recipe to include the new outouts @mediocretech pointed out (athena & databricks)
We need the pending admin request to be merged - https://github.com/conda-forge/admin-requests/pull/1383

Thank you and regards.

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13285460037](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13285460037). Examine the logs at this URL for more detail.</sub>

rxm7706 committed:
Trigger Rerun

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13285623563](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13285623563). Examine the logs at this URL for more detail.</sub>

rxm7706 commented:
> @rxm7706 Is this ready for merge?

@cpcloud - This is now ready to review & Merge. 
Thank you and Regards.

gforsyth reviewed (approved):
Excellent work @rxm7706 -- thanks for taking this on.
